### PR TITLE
refactor: move from github.com/pkg/errors to 'errors' and 'fmt' packages

### PR DESCRIPTION
**Pull request by mmorel-35**
The package https://github.com/pkg/errors is not maintained anymore.
This PR replaces https://github.com/pkg/errors with official errors and fmt golang packages

Signed-off-by: Matthieu MOREL <mmorel-35@users.noreply.github.com>

### CODE REVIEW

File: cmd/promtool/backfill.go [hunk 1]
```diff
@@ -218,7 +219,7 @@ func backfill(maxSamplesInAppender int, input []byte, outputDir string, humanRea
 	p := textparse.NewOpenMetricsParser(input)
 	maxt, mint, err := getMinAndMaxTimestamps(p)
 	if err != nil {
-		return errors.Wrap(err, "getting min and max timestamp")
+		return fmt.Errorf("getting min and max timestamp: %w", err)
 	}
-	return errors.Wrap(createBlocks(input, mint, maxt, int64(maxBlockDuration/time.Millisecond), maxSamplesInAppender, outputDir, humanReadable, quiet), "block creation")
+	return fmt.Errorf("block creation: %w", createBlocks(input, mint, maxt, int64(maxBlockDuration/time.Millisecond), maxSamplesInAppender, outputDir, humanReadable, quiet))
```
Comment (roidelapluie) [inline cmd/promtool/backfill.go:224]: This seems like a brekaking change. Error.Wrap will not return an error if err is nil.

Comment (roidelapluie) [reply cmd/promtool/backfill.go:224]: it's probably easier for review do do this per directory rather than a bit all in one pull request

**Comment by roidelapluie** [comment]
it's probably easier for review do do this per directory rather than a bit all in one pull request

**Comment by mmorel-35** [comment]
Ok! I should probably close this PR and create one per folder. Which one should be first?

**Comment by roidelapluie** [comment]
this is up to you :)

**Comment by mmorel-35** [comment]
Replaced by : 
* #10655

**Pull request closed without merging**